programs/loader-v4/src/lib.rs

```diff
@@ -488,12 +484,22 @@ pub fn process_instruction_retract(
         );
         return Err(InstructionError::InvalidArgument);
     }
-    if matches!(state.status, LoaderV4Status::Retracted) {
+    if !matches!(state.status, LoaderV4Status::Deployed) {
         ic_logger_msg!(log_collector, "Program is not deployed");
```

Comment:
The log message should also be changed, as the status can be `Finalized`.

Reply:
The `check_program_account()` above prevents the `Finalized` case.